text/0028-bill-of-materials.md

```diff
@@ -0,0 +1,166 @@
+# Implement a Bill of Materials Across Paketo
+
+## Summary
+
+A bill of materials (BOM) is a set of metadata that provides an account of the
+dependencies and tools that are both present in an application image, as well
+those used in the construction of that image. The availability of this data is
+a key value proposition of Cloud Native Buildpacks.
+
+As an implementation of Cloud Native Buildpacks, the Paketo project should
+support adding bill of materials metadata across the different components that
+go into an app image. This RFC serves as an intent to implement the BOM and
+defines the minimal set of key/values to be included.
+
+## Motivation
+
+A BOM can include an array of different metadata about application dependencies
+and buildpack dependencies such as SHAs, URIs, versions, license information,
+vulnerability identifiers, etc. This information is valuable for users who want
+an accurate inventory of everything in their final application image, or that
+may have been used to build it. Having access to that information makes it
+easier to audit for vulnerabilities and license information.
+
+The Paketo Java Buildpacks as well as a handful of other Paketo Buildpacks
+already make the BOM available. Another motivation for doing this work is to
+achieve a consistent user experience across the Paketo project.
+
+## Detailed Explanation
+
+The following Paketo components should have BOM metadata attached to them:
+
+### Overall Schema
+The bill of materials for each of type of entry should have some subset of
+fields that conform to the overall schema below.
+```
+[[bom]]
+name = "<dependency name>"
+
+[bom.metadata]
+  version = "<dependency version>"
+  sha256 = "<hash of dependency artifact from uri>"
+  uri = "<uri to dependency>"
+  arch = "<compatible architecture>"
+  summary = "<package summary>"
+  cpe = "<version-specific common platfrom enumeration>"
+  purl = "<package URL>"
+  licenses = [<licenses that the dependency has>]
+
+[bom.metadata.source]
+  name = "<dependency source name>"
+  version = "<dependency source version>",
+  uri = "<uri to the dependency source>"
+  sha256 = "<hash of the dependency source artifact from source-uri>"
+  upstream-version = "<dependency source upstream version>"
+```
+
+### Stacks
+Stacks (such as those found in the [Paketo Stacks
+repository](https://github.com/paketo-buildpacks/stacks)) should have BOM
+metadata that includes in-depth information on all of the OS level packages
+installed as part of the stack.
+
+The minimal set of keys to include in these type of BOM entries are:
+```
+[[bom]]
+name = "<dependency name>"
+
+[bom.metadata]
+  version = "<dependency version>"
+
+[bom.metadata.source]
+  name = "<dependency source name>"
```

Comment:
```suggestion
  name = "<dependency source name>"
  purl = "<package URL>"
```

Given all of the stack dependencies are ubuntu LTS packages it should be easy to provide valid purls for all of these. e.g.
```
pkg:deb/ubuntu/curl@7.50.3-1?arch=amd64&distro=bionic
```